### PR TITLE
Rename SearchIndexerDataSource to SearchIndexerDataSourceConnection

### LINKED ISSUE #12350: Rename SearchIndexerDataSource to SearchIndexerDataSourceConnection
Missed this for #12217 and needs to be done now.

Also rename `SearchIndexerClient` methods from "DataSource" to "DataSourceConnection".

### PR DESCRIPTION
**Pull request by heaths**
Fixes #12350. Also got a few updates from swagger.

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Indexes/Models/SearchIndexerDataSource.cs [hunk 1]
```diff
@@ -6,21 +6,22 @@
 
 namespace Azure.Search.Documents.Indexes.Models
 {
-    [CodeGenSuppress(nameof(SearchIndexerDataSource), typeof(string), typeof(SearchIndexerDataSourceType), typeof(DataSourceCredentials), typeof(SearchIndexerDataContainer))]
-    public partial class SearchIndexerDataSource
+    [CodeGenModel("SearchIndexerDataSource")]
+    [CodeGenSuppress(nameof(SearchIndexerDataSourceConnection), typeof(string), typeof(SearchIndexerDataSourceType), typeof(DataSourceCredentials), typeof(SearchIndexerDataContainer))]
+    public partial class SearchIndexerDataSourceConnection
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/Indexes/Models/SearchIndexerDataSource.cs:11]: Nit - rename the file as well?

**Review by tg-msft — COMMENTED** [review]
Looks good

File: sdk/search/Azure.Search.Documents/src/Indexes/Models/SearchIndexerDataSource.cs (hunk 1 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Indexes/Models/SearchIndexerDataSource.cs:11]: Good catch. Forgot, but fixed now.

**Review by tg-msft — APPROVED** [review]

**Pull request merged**